### PR TITLE
[CurrencyService] Add C++ Metrics

### PR DESCRIPTION
**Pull request by julianocosta89**
# Changes

This PR adds the manual metric `app_currency_counter` to the currency service.
The metric counts how many times a specific currency was converted.

image

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/

### CODE REVIEW

**Comment by julianocosta89** [comment]
@lalitb would you mind taking a look at this one?

**Comment by cartersocha** [comment]
Could you open and link a pr to the website docs adding this as a user created metric ?

**Comment by julianocosta89** [comment]
> Could you open and link a pr to the website docs adding this as a user created metric ?

Done!

File: src/currencyservice/src/meter_common.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+#include "opentelemetry/exporters/otlp/otlp_grpc_metric_exporter_factory.h"
+#include "opentelemetry/metrics/provider.h"
+#include "opentelemetry/sdk/metrics/aggregation/default_aggregation.h"
```
Comment (lalitb) [inline src/currencyservice/src/meter_common.h:18]: nit - do we need this header ? Probably I need to fix the example code in otel-cpp repo too.

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for enabling C++ metrics in demo.

**Review by mviitane — APPROVED** [review]
Great improvement! This runs fine.

File: src/currencyservice/src/meter_common.h (hunk 1 above)
Comment (julianocosta89) [reply src/currencyservice/src/meter_common.h:18]: Thanks for the review @lalitb!

I've removed:
```
#include <cstdint>
#include "opentelemetry/sdk/metrics/aggregation/default_aggregation.h"
```

Without the other 2 the build failed, so I let them in.

File: src/currencyservice/src/meter_common.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace
+{
+  std::string version{ "1.3.0" };
+  std::string name{ "app_currency" };
+  std::string schema{ "https://opentelemetry.io/schemas/1.2.0" };
+
+  void initMeter() 
+  {
+    //Build MetricExporter
```
Comment (mic-max) [inline src/currencyservice/src/meter_common.h:36]: nit: space after `//` comments here, 39, and 44

**Pull request merged**